python/tests/test_variable.py

```diff
@@ -551,11 +551,14 @@ def test_make_variable_from_unit_scalar_mult_div():
     var.unit = sp.units.m**(-1)
     assert var == 0.0 / sp.units.m
 
-    # TODO change next string with normal constructor
-    # for 0D Variable the it introduced
-    v = sp.Variable([sp.Dim.X], values=np.array([0]), dtype=np.float32)
-    var = sp.Variable(v[sp.Dim.X, 0])
+    var = sp.Variable(np.float32())
     var.unit = sp.units.m
     assert var == np.float32(0.0) * sp.units.m
     var.unit = sp.units.m**(-1)
     assert var == np.float32(0.0) / sp.units.m
+
+
+def test_construct_0d_numpy():
+    v = sp.Variable([sp.Dim.X], values=np.array([0]), dtype=np.float32)
```

Comment:
Is this test sufficient? Unless I misunderstood, arrays always worked as expected and the issue only arose when creating a `Variable` from a single Numpy scalar type.

Reply:
> Is this test sufficient? Unless I misunderstood, arrays always worked as expected and the issue only arose when creating a `Variable` from a single Numpy scalar type.

Yes, exactly.

Reply:
But that isn't what is being tested here, right? Should the test not be pretty much the code listed in the description of #571?

Reply:
It is the same to lines below `assert var == sp.Variable(np.float32())`

Reply:
Ah, yes I see now.
Is there a reason you don't just do, to me this makes what you are actually testing clearer:
```python
v = sc.Variable(np.float32())
assert v.dtype == sc.dtype.float
```